### PR TITLE
Re-add code (for code review)

### PR DESCRIPTION
**Pull request by yotammanor**
Reverts Orisbaum/TIC-TAC-TOE#1

(Adds the code)

### CODE REVIEW

File: tik_tack_toe/tick_tack_toe.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+
+def check_board(board):
+    i=0
+    free_cells=[]
+    for i in xrange(len(board)):
+        if board[i] == '':
+            free_cells.append(i+1)
+        i+=1
+    return free_cells
```
Comment (yotammanor) [inline tik_tack_toe/tick_tack_toe.py]: The pythonic way to write this function is like this::
```
def check_board(board):
    return [cell for cell in board if cell]
```
Let's explain what we have here.
This is what's called **List comprehension**.  This allows us to replace this:
```
some_list = []
for cell in board:
    if cell:
        some_list.append(cell)
```

With this:
```
some_list = [cell for cell in board if cell]
```

Note that the syntax for list comprehension is reminiscent of a mathematical definition of a set (i.e. `{x|x in X|if ...}`), this is on purpose, of course.

It's a bit hard to get used to list comprehension at the beginning, but once you do, you won't be able to use anything else. :)

Another thing we can see is that in python you don't have to define an index variable (`i` in your case), you can directly **iterate** on the items in the given list.

So the expression:
```
i = 0
for i in xrange(len(board)):
    print board[i]
```
is the same as:
```
for cell in board:
    print cell
```

Note that it also solves the problem that the board[0]=1, board[1] =2, etc. You directly iterate over the cells, so you can use the values as they are.

**To sum up** - I took 7 lines of code, turned them into 1 line of code, and while doing so I did not loose any of the clarity of the code (and I might have increased it, especially if you're familiar with list comprehension). This is the power of python.

File: tik_tack_toe/tick_tack_toe.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        put_on_board_y(board, p2)
+        winner = check_win(board)
+        print_board(board)
+        if winner:
+            print('%s is the winner!'%winner)
+            break
+
+def put_on_board_x(board,p1):
+    p1-=1
+    board[p1]='X'
```
Comment (yotammanor) [inline tik_tack_toe/tick_tack_toe.py]: I'd write it `board[p1 - 1] = 'X'`, I think it's clearer this way.

File: tik_tack_toe/tick_tack_toe.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    return board
+
+def put_on_board_y(board,p2):
+    p2-=1
+    board[p2]='O'
+    return board
+
+def print_board(board):
+    i=0
+    new_board=['','','','','','','','','']
```
Comment (yotammanor) [inline tik_tack_toe/tick_tack_toe.py]: In python you can write this line like this:
`new_board = [''] * 9`

Note that besides being easier to write, it also exposes your intent, and helps you avoid mistakes and/or spotting a mistake more easily.

File: tik_tack_toe/tick_tack_toe.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    for i in xrange(len(board)):
+        if board[i] != '':
+            new_board[i]=board[i]
+        else:
+            new_board[i] = i+1
+    print('%s | %s | %s\n%s | %s | %s\n%s | %s | %s'%(new_board[0],new_board[1],new_board[2],new_board[3],new_board[4],new_board[5],new_board[6],new_board[7],new_board[8]))
```
Comment (yotammanor) [inline tik_tack_toe/tick_tack_toe.py]: This line can be written as such:

```
print ('{}|{}|{}\n{}|{}|{}\n{}|{}|{}'.format(*new_board))
```

Note the following:

1. I'm using a different syntax for string formatting. instead of `%s` I have `{}`, and instead of `% (<values>)` I am using `.format(<values>)`. This is the newer syntax, and it is recommended to use this instead of the older one. It has more capabilities, especially since it is a done using a function, instead of a unique operator (the `%`). Since you are new, you can start with the new one write now!
2. One of the things that this syntax allows is to send the entire board, and expand it two nine variables, by using the `*` prefix. Typing `.format(*new_board)` is the same is typing down `.format(new_board[0],...,new_board[8])`, but obviously faster, less error-prone, and easier to read once you know the trick.
